backend/corpora/common/entities/dataset_asset.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import typing
+
+import boto3
+from botocore.exceptions import ClientError
+
+from .entity import Entity
+from ..corpora_orm import DbDatasetArtifact
+from urllib.parse import urlparse
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetAsset(Entity):
+    table = DbDatasetArtifact
+    s3 = boto3.client("s3")
+
+    def __init__(self, db_object: DbDatasetArtifact):
+        super().__init__(db_object)
+
+        url = urlparse(self.s3_uri)
+        self.bucket_name = url.netloc
+        self.key_name = url.path[1:]
```

Comment:
Consider `url.path.lstrip("/")` just in case it's empty.

Reply:
Eh probably never mind, if it's empty it's just going to break immediately elsewhere.